pkg/promotion/promotion.go

```diff
@@ -121,11 +123,20 @@ func (o *Options) Validate() error {
 
 func (o *Options) Bind(fs *flag.FlagSet) {
 	fs.StringVar(&o.CurrentRelease, "current-release", "", "Configurations targeting this release will get branched.")
+	fs.StringVar(&o.CurrentPromotionNamespace, "current-promotion-namespace", ocpPromotionNamespace, "The promotion namespace of the current release.")
 	o.ConfirmableOptions.Bind(fs)
 }
 
 func (o *Options) matches(configuration *cioperatorapi.ReleaseBuildConfiguration) bool {
```

Comment:
Could we write something like:

```go
var imagesMatch bool
if o.CurrentPromotionNamespace == "" {
    imagesMatch = PromotesOfficialImages(configuration)
} else {
    imagesMatch = PromotesImagesInto(configuration, o.CurrentPromotionNamespace)
}

return imagesMatch && configuration.PromotionConfiguration.Name == o.CurrentRelease
```

Where we can factor `PromotesOfficialImages` to use `PromotesImagesInto` with `"ocp"` constant and keep the business logic for this in promotion.go

Reply:
yes, I can introduce the `PromotesImagesInto`  function and this makes the logic more clear